gems/pending/spec/postgres_ha_admin/failover_monitor_spec.rb

```diff
@@ -0,0 +1,135 @@
+      before do
+        allow(PG::Connection).to receive(:open).and_return(connection)
+      end
+
+      it "updates 'failover_databases.yml'" do
+        expect(failover_db).to receive(:update_failover_yml)
+        failover_monitor.monitor
+      end
+
+      it "does not stop evm server and does not execute failover" do
+        expect(failover_db).to receive(:update_failover_yml)
+        expect(linux_admin).not_to receive(:stop)
+        expect(linux_admin).not_to receive(:restart)
+        expect(failover_monitor).not_to receive(:execute_failover)
+
+        failover_monitor.monitor
+      end
+    end
+
+    context "primary database is not accessable" do
+      before do
+        allow(PG::Connection).to receive(:open).and_return(nil, connection, connection)
+        stub_monitor_constants
+      end
+
+      it "stop evm server(if it is running) before failover attempt" do
+        expect(linux_admin).to receive(:stop).ordered
+        expect(failover_monitor).to receive(:execute_failover).ordered
+        failover_monitor.monitor
+      end
+
+      it "does not update 'database.yml' and 'failover_databases.yml' if all standby DBs are in recovery mode" do
+        failover_not_executed
+        expect(PostgresAdmin).to receive(:database_in_recovery?).and_return(true, true, true).ordered
+        failover_monitor.monitor
+      end
+
+      it "does not update 'database.yml' and 'failover_databases.yml' if there is no master database avaiable" do
+        failover_not_executed
+        expect(PostgresAdmin).to receive(:database_in_recovery?).and_return(false, false, false).ordered
+        expect(failover_db).to receive(:host_is_repmgr_primary?).and_return(false, false, false).ordered
+        failover_monitor.monitor
+      end
+
+      it "updates 'database.yml' and 'failover_databases.yml' and restart evm server if new primary db available" do
+        failover_executed
+        allow(PostgresAdmin).to receive(:database_in_recovery?).and_return(false)
+        allow(failover_db).to receive(:host_is_repmgr_primary?).and_return(query_repmanager_result)
+        failover_monitor.monitor
+      end
+    end
+  end
+
+  def failover_not_executed
+    expect(linux_admin).to receive(:stop)
+    expect(failover_db).to receive(:active_databases).and_return(active_databases_list)
+    expect(failover_db).not_to receive(:update_failover_yml)
+    expect(db_yml).not_to receive(:update_database_yml)
+    expect(linux_admin).not_to receive(:restart)
+  end
+
+  def failover_executed
+    expect(linux_admin).to receive(:stop)
+    expect(failover_db).to receive(:active_databases).and_return(active_databases_list)
+    expect(failover_db).to receive(:update_failover_yml)
+    expect(db_yml).to receive(:update_database_yml)
+    expect(linux_admin).to receive(:restart)
+  end
+
+  def failover_not_executed
+    expect(linux_admin).to receive(:stop)
+    expect(failover_db).to receive(:active_databases).and_return(active_databases_list)
```

Comment:
Because these two lines are common it might make sense to put them in the `before` block of the "primary database is not accessable" context.